src/types/graphql_type_int.erl

```diff
@@ -0,0 +1,39 @@
+-module(graphql_type_int).
+
+-export([
+  type/0
+]).
+
+type()-> #{
+  kind => 'SCALAR',
+  name => 'Int',
+  ofType => null,
+  description => <<
+    "The `Int` scalar type represents non-fractional signed whole numeric ",
+    "values. Int can represent values between -(2^53 - 1) and 2^53 - 1 since ",
+    "represented in JSON as double-precision floating point numbers specified ",
+    "by [IEEE 754](http://en.wikipedia.org/wiki/IEEE_floating_point)."
+  >>,
+
+  serialize => fun serialize/1,
+  parse_value => fun parse_value/1,
+  parse_literal => fun parse_literal/1
+}.
+
+serialize(Value) -> coerce(Value).
+
+parse_value(Value) -> coerce(Value).
+
+-spec parse_literal(map()) -> integer().
+parse_literal(#{kind := Kind, value := Value}) when
+  Kind =:= 'IntValue' ->
+    coerce(Value);
+parse_literal(#{kind := Kind}) ->
+  throw({error, type_validation, <<"Unexpected value type. Got: ", (atom_to_binary(Kind, utf8))/binary, ", expected: IntValue">>}).
+
+
+%% TODO: add binary, string and float representation
+-spec coerce(null | integer()) -> integer().
+coerce(null) -> null;
+coerce(Value) when is_integer(Value) -> Value;
```

Comment:
GraphQL servers should coerce non‐int raw values to Int when possible otherwise they must raise a field error. Examples of this may include returning 1 for the floating‐point number 1.0, or 2 for the string "2".